core/src/packet.rs

```diff
@@ -60,6 +68,12 @@ impl Default for Packet {
     }
 }
 
+impl PartialEq for Packet {
+    fn eq(&self, other: &Packet) -> bool {
+        self.data.iter().zip(other.data.iter()).all(|(a, b)| a == b) && self.meta == other.meta
```

Comment:
Can you add some comments explaining why the default implementation of `PartialEq` is inadequate here?

Reply:
There is no default implementation of PartialEq for fixed size arrays of size > 32